front/js/cart.js

```diff
@@ -154,3 +154,90 @@ const deleteItem = () => {
+const addressRegex = /^[A-Za-zÀ-ÿ0-9'-\s]{2,50}$/; //all letters & number, characters ' - and space from 0-50 characters
+const villeRegex = /^[A-Za-z'0-9'-\s]{2,30}$/;
+
+//  function to check the regex and return msg
+function formIsValide(input, regex, message) {
+  let testRegex = regex.test(input.value);
+  if (!testRegex) {
+    document.getElementById(`${input.id}ErrorMsg`).innerHTML = message;
+    return false;
+  } else {
+    document.getElementById(`${input.id}ErrorMsg`).innerHTML = "";
+    return true;
+  }
+}
+//get parent and messages error
+const form = document.querySelector(".cart__order__form");
+const firstAndLastNameMessage =
+  "Votre saisie n'est pas valide, veuillez saisir au moin 2 caractères. Les numéros ne sont pas acceptés.";
+const villeMessage = "Veuillez saisir votre ville";
+const addressMessage = "Veuillez saisir votre address";
+const emailMessage =
+  "Veuillez saisir votre email correctement. exp: abc@domain.com";
+
+//submit form
+form.addEventListener("submit", (event) => {
+  debugger;
+  event.preventDefault();
+  //declare info of client get from forms
+  let contact = {
+    firstName: form.firstName.value,
+    lastName: form.lastName.value,
+    address: form.address.value,
+    city: form.city.value,
+    email: form.email.value,
+  };
+
+  //get array of ids in cart
+  let products = [];
+  cartItems.forEach((item) => {
+    products.push(item.id);
+  });
+
+  //condition is check all forms are correctly fullfilled
+  if (
+    formIsValide(form.firstName, nameRegex, firstAndLastNameMessage) &&
+    formIsValide(form.lastName, nameRegex, firstAndLastNameMessage) &&
+    formIsValide(form.address, addressRegex, addressMessage) &&
+    formIsValide(form.city, villeRegex, villeMessage) &&
+    formIsValide(form.email, emailRegex, emailMessage)
+  ) {
+    //send client info & ids to serveur by POST
+    fetch(`http://localhost:3000/api/products/order`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ contact, products }),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+      })
+      //add data to local storage and direct to page confirmation
+      .then((data) => {
+        localStorage.setItem("order", JSON.stringify(data));
+        window.location.href = `./confirmation.html`;
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    //call function to show message errs if forms aren't correctly filled
+    formIsValide(form.firstName, nameRegex, firstAndLastNameMessage);
+    formIsValide(form.lastName, nameRegex, firstAndLastNameMessage);
+    formIsValide(form.address, addressRegex, addressMessage);
+    formIsValide(form.city, villeRegex, villeMessage);
+    formIsValide(form.email, emailRegex, emailMessage);
```

Comment:
To avoide this, ou can set a let errors before formIsValide function, then of the function replace return false by errors=false and return true by errors=true. Then call the 5 formIsValid before the if line 206 and on the if write if(!errors).